### DISCUSSION TITLE
Search for request-id (and other log fields) in Grafana Loki

### DISCUSSION THREAD

**Discussion by theliuk** [mia-platform/community#172] [Q&A] [answered]
When we used Kibana, we were able to look for a specific `requestId` (I believe we had an index on this log field).
How can we do it in Grafana Loki? Cause it seems to me there's no index on this log field.
It isn't a label, and the only way I have to search for a request-id is to use the `Line contains` operator
<img width="785" alt="Schermata 2023-01-16 alle 10 57 07" src="https://user-images.githubusercontent.com/809858/212653258-b025430f-0f0f-4c6d-9727-99f31a27ed99.png">

But there's a problem: if I try to search for a request-id far in the past (more than 7 days ago), the query takes a lot of time and eventually fails because of timeout. 

The same goes for other fields that used to be indexed in Kibana, like the `httpStatusCode`.

We need to perform queries on those fields efficiently. What are we missing?

**Comment by davidebianchi** [comment]
Hi @theliuk, what type of query are you doing?
Can you try something like:

```
{k8sCluster="YOUR_CLUSTER"} | json | reqId = "THE_REQID"
```

**Reply by theliuk** [reply]
Same result.
By reading this [wiki](https://makeitapp.atlassian.net/wiki/spaces/MP/pages/3743383572/Loki+migration), I believe `reqId` isn't a detected field.
<img width="1164" alt="Schermata 2023-01-16 alle 11 17 57" src="https://user-images.githubusercontent.com/809858/212654458-b362271d-9da2-4c62-81a0-0a03dada9915.png">

**Reply by davidebianchi** [reply]
It is not important, the query inside loki does not need write indexes. Can you try to reduce the time range of the query to speedup the search, but it is strange because usually I reach this behaviour when I query for 1 month.
If you reduce the scope adding a namespace?

```
{k8sCluster="YOUR_CLUSTER", namespace="YOUR_NAMESPACE"} | json | reqId = "THE_REQID"
```

**Reply by theliuk** [reply]
I need to look past 7 days. 
Same problem, even if a reduce the scope to the container level.

**Reply by theliuk** [reply]
Now it worked (maybe there was a high demand previously).
Anyhow, are we saying we can't look for detected fields with a time span of 30 days? Because it really looks like a serious limitation.

**Reply by davidebianchi** [reply, marked as answer]
It depends on the amount of logs I suppose. Once timeout, it needs to have a more strict query to perform. Maybe it is possible to increase the resources

**Reply by theliuk** [reply]
Ok, thanks
